authz/src/main/java/io/grpc/authz/FileWatcherAuthorizationServerInterceptor.java

```diff
@@ -93,22 +76,25 @@ void updateInternalInterceptor() throws IOException {
    * 
    * @param period the period between successive file load executions.
    * @param unit the time unit for period parameter
+   * @param executor the execute service we use to read and update authorization policy
    * @return an object that caller should close when the file refreshes are not needed
    */
-  public Closeable scheduleRefreshes(long period, TimeUnit unit) 
-    throws IOException {
+  public Closeable scheduleRefreshes(
+      long period, TimeUnit unit, ScheduledExecutorService executor) throws IOException {
+    checkNotNull(executor, "scheduledExecutorService");
     if (period <= 0) {
       throw new IllegalArgumentException("Refresh interval must be greater than 0");
     }
     final ScheduledFuture<?> future = 
-        scheduledExecutorService.scheduleWithFixedDelay(new Runnable() {
-          @Override public void run() {
+        executor.scheduleWithFixedDelay(new Runnable() {
+          @Override
+          public void run() {
             try {
-                updateInternalInterceptor();
+              updateInternalInterceptor();
             } catch (Exception e) {
-                logger.log(Level.WARNING, "Authorization Policy file reload failed: " + e);
-            }
+              logger.log(Level.WARNING, "Authorization Policy file reload failed: ", e);
```

Comment:
Remove ": " from string. `e` is not a formatted argument. It is its own separate thing, displayed however the logging framework wants.